rfcs/011-resource-access-control/011-resource-access-control.md

```diff
@@ -6,136 +6,328 @@
 - there is a default resources pool that all users have access to
 - resource pools have quotas on how much total resources are available in them
-- resource pools contain different flavours/combinations of preset gpu, cpu, memory and storage
-- convenient and transparent way for admins to define resource pools and assign users
+- resource pools contain different flavors/combinations of preset GPU, CPU, memory and storage
+
+Nice to have requirements:
+- a convenient and transparent way for administrators to define resource pools and assign users
 
 ## Key Assumptions
 
-- resource pools are assigned on a per user basis
-- Renku admins are in control of creating resource pools and assigning users to them
-- information about resource pools is persisted in a database and controled by a 
-  dedicated service
+- A **default** resource pool always exists and all users can access it
+- Resource pools are assigned on a per-user basis
+- Renku administrators are in control of creating resource pools and assigning users to them
+- Resource pools are created through Renkulab which performs some validation on their configuration
+- A Renkulab administrator is a Keycloak user that belongs to the `admin` group
 
-## Possible Solutions
+## Possible solutions
 
 - store all information in Keycloak
 - store information in a database with a dedicated service to modify the data
-- store all information only in k8s manifests/configmaps
+- store all information only in Kubernetes manifests/ConfigMaps
 
 ## Proposed solution
 
+This RFC proposes to solve the stated problem by having the user's request to launch a session pass through two gates:
+ 1. Does the user have access to the selected resource pool?
+ 2. Does the selected resource pool have enough quota to accept the user request?
+
+Each of those two gates is implemented by different services, the first is an in-house service that exposes through an
+HTTP interface if a specific user is granted access to a resource pool, referred from here on as the **Resource Access Control**
+service, with the data kept in a Postgres database.
+The second gate is implemented through [Kubernetes resource quotas](https://kubernetes.io/docs/concepts/policy/resource-quotas/)
+to define the resource pools and the [resource quota per PriorityClass](https://kubernetes.io/docs/concepts/policy/resource-quotas/#resource-quota-per-priorityclass)
+to associate user sessions to resource pools.
+
 ![Basic architecture diagram](./overall-flow.png)
 
-Store information about resource pools in postgres. Create a dedicated service which will
-expose an API that will allow admins to create resource pools, modify them or assign users
-to different resource pools.
+### The Resource Access Control Service
+
+The service exposes an HTTP API that the administrators interact with to create/update/delete resource pools and
+add/remove users to resource pools.
+
+A prototype of the API exposed by the service looks like the following:
+```yaml
```

Comment:
Can we please add this as a separate file in the same folder as the markdown file? Then we can simply refer to it via a link or call it out. There is no reason to embed the whole spec directly in the already fairly long markdown file.